pkg/apis/pubsub/v1alpha1/pull_subscription_types.go

```diff
@@ -94,22 +94,27 @@ type PullSubscriptionSpec struct {
 	// +optional
 	RetentionDuration *time.Duration `json:"retentionDuration,omitempty"`
 
-	// Sink is a reference to an object that will resolve to a domain name to
-	// use as the sink.
+	// Sink is a reference to an object that will resolve to a domain name or a
+	// URI directly to use as the sink.
 	// +optional
-	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+	Sink *Destination `json:"sink,omitempty"`
```

Comment:
Why is this optional? Just curious.

Reply:
Looking at the validation code, doesn't seem so optional? :)

Reply:
nice catch! it should not be optional.

Reply:
If it should not be optional, why not:
Sink Destination `json:"sink"`